wallet-new/src/Cardano/Wallet/API/V1/Internal.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+-- | This module contains the top level API definition for frontend-related
+-- tasks.  The API endpoints presented here are intended for use with the
+-- Daedalus client, and aren't useful for wallets, exchanges, and other users.
+module Cardano.Wallet.API.V1.Internal where
+
+import           Universum
+
+import           Servant
+
+import           Cardano.Wallet.API.Response
+import           Cardano.Wallet.API.V1.Types
+import           Pos.Util.Mnemonic
+
+-- migrate everything except for import/export
+
+type API =
+    "internal"
+        :> ( "update"
+            :> ( "apply"
+                :> Post '[ValidJSON] NoContent
+            :<|> "postpone"
+                :> Post '[ValidJSON] NoContent
+            )
+        :<|> "redemptions"
+            :> "ada"
+            :> QueryParam "paper_vend_passphrase" (Mnemonic 9)
+            :> ReqBody '[ValidJSON] WalletRedemption
+            :> Get '[ValidJSON] (WalletResponse Transaction)
+        )
+
+newtype Seed = Seed Text
+    deriving (Eq, Show, Generic)
+
+data WalletRedemption = WalletRedemption
+    { walletRedemptionWalletId :: WalletId
+    , walletRedemptionSeed     :: Seed
+    } deriving (Eq, Show, Generic)
+
+-- redeemAda
+-- ProtocolMagic -> (TxAux -> m Bool) -> PassPhrase -> CWalletRedeem -> m CTx
+--
+-- -- | Query data for redeem
+-- data CWalletRedeem = CWalletRedeem
+--     { crWalletId :: !CAccountId
+--     , crSeed     :: !Text -- TODO: newtype!
+--     } deriving (Show, Generic)
+
+-- redeemAdaPaperVend
+--     :: MonadWalletTxFull ctx m
+--     => ProtocolMagic
+--     -> (TxAux -> m Bool)
+--     -> PassPhrase
+--     -> CPaperVendWalletRedeem
+--     -> m CTx
+-- data CPaperVendWalletRedeem = CPaperVendWalletRedeem
+--     { pvWalletId     :: !CAccountId
+--     , pvSeed         :: !Text -- TODO: newtype!
+--     , pvBackupPhrase :: !(CBackupPhrase 9)
+--     } deriving (Show, Generic)
+
+-- The redemption datatypes differ only by the backup phrase. Otherwise the
+-- wallet ID and seed are both present. Potentially we can unify these into the
+-- same endpoint. Seems like a query parameter like `paper_vend_passphrase=...`
+-- would do nicely.
```

Comment:
I agree. I see no point for two endpoints here. I'd rather unify them and have a switch kind of parameter to target the right behavior. Perhaps it's not even needed and we can, at parsing-time, identify which structure is being passed and pick the right behavior after that.

Reply:
Note that, I am not sure this `redeem` should be in the _internal_ part of the API. We probably want to have them in the public-facing part?